Firestore/Example/Podfile

```diff
@@ -39,34 +79,20 @@ target 'Firestore_Example_iOS' do
   pod 'Firebase/CoreOnly', '6.6.0'
 
   pod 'FirebaseFirestore', :path => '../../'
+  configure_local_pods()
```

Comment:
I think putting this here breaks the Swift tests

Reply:
What broke the Swift tests was actually removing the local pod for FirebaseFirestoreSwift. I've added this to `configure_test` and that resolves the build issue.

I've also merged #3647, cleaned this up, and added comments. PTAL.

Note that handling FirebaseFirestoreSwift uniformly resulted in a diff to the project.